tests/snappi_tests/dash/ha/ha_helper.py

```diff
@@ -0,0 +1,861 @@
+        stats_client.append(res)
+        time.sleep(60)
+
+        # req1 = api.metrics_request()
+        # req1.choice= "httpserver"
+        # req1.httpserver.stat_name = ["TCP Connections in ESTABLISHED State", "TCP FIN Received","HTTP Bytes Received"]
+        # #req1.httpserver.all_stats=True # for all stats - True
+        # res1 = api.get_metrics(req1).httpserver_metrics
+        # logger.info("#### res1 = {} ####".format(res1))
+
+        cps_max = 0
+        client_stat_values = []
+        for stat in stats_client:
+            tmp = re.findall(r"value: '(\d+)'", str((stat)))
+            client_stat_values += tmp
+            client_stat_values = [int(item) for item in client_stat_values]
+        cps_max = max(client_stat_values)
+
+        if cps_max < test_value:
+            test = False
+        else:
+            test = True
+
+        if test:
+            logger.info('Test Iteration Pass')
+            test_result = "Pass"
+            MIN_CPS = test_value
+            test_value = (MAX_CPS + MIN_CPS) / 2
+        else:
+            logger.info('Test Iteration Fail')
+            test_result = "Fail"
+            MAX_CPS = test_value
+            test_value = (MAX_CPS + MIN_CPS) / 2
+
+        columns = ['#Run', 'CPS Objective', 'Max CPS', 'Test Result']
+        testRuns.append([test_iteration, cps_objective_value, cps_max, test_result])
+        table = tabulate(testRuns, headers=columns, tablefmt='psql')
+        logger.info(table)
+
+        logger.info("Iteration Ended...")
+        logger.info('MIN_CPS = %d' % MIN_CPS)
+        logger.info('Current MAX_CPS = %d' % MAX_CPS)
+        logger.info('Previous CPS Objective value = %d' % old_value)
+        logger.info(' ')
+        test_iteration += 1
+        logger.info("Releasing config...")
+        try:
+            # Code that may raise an exception
+            param = {}
+            res = api.ixload_configure("post", releaseConfig_url, param)
+        except Exception as e:
+            # Handle any exception
+            logger.info(f"An error occurred: {e}")
+        logger.info("Releasing config completed..")
+
+        logger.info("Changing app state to stop")
+        cs.app.state = 'stop'  # cs.app.state.START
+        api.set_control_state(cs)
+
+    return api
+
+
+"""
+def test_saveAs(api, test_filename):
+
+    saveAs_operation = 'ixload/test/operations/saveAs'
+    #url = "{}/{}".format(base_url, saveAs_operation)
+    paramDict = {
+        'fullPath': "C:\\automation\\{}.rxf".format(test_filename),
+        'overWrite': True
+    }
+
+    #response = requests.post(url, data=json.dumps(paramDict), headers=headers)
+    try:
+        # Code that may raise an exception
+        res = api.ixload_configure("post", saveAs_operation, paramDict)
+    except Exception as e:
+        # Handle any exception
+        logger.info(f"An error occurred: {e}")
+
+    return
+"""
+
+
+def main(connection_dict, test_type, test_filename, initial_cps_value):
```

Comment:
this `main` method actually generates ixload traffic right? maybe change to a more specific method name?